format.go

```diff
@@ -533,32 +534,58 @@ func parseURL(s string) (*gourl.URL, error) {
 }
 
 func validateURI(v any) error {
+	return validateRI(v, false, false)
+}
+
+func validateIRI(v any) error {
+	return validateRI(v, true, false)
+}
+
+func validateURIReference(v any) error {
+	return validateRI(v, false, true)
+}
+
+func validateIRIReference(v any) error {
+	return validateRI(v, true, true)
+}
+
+func validateRI(v any, international bool, reference bool) error {
 	s, ok := v.(string)
 	if !ok {
 		return nil
 	}
+
+	if !international {
+		// rfc3986: 2.  Characters
+		//   [..] The ABNF notation defines its terminal values to be
+		//   non-negative integers (codepoints) based on the US-ASCII coded
+		//   character set
+		for _, r := range s {
```

Comment:
Maybe use `if strings.ContainsFunc(s, func(r rune) bool { return r > unicode.MaxASCII }) {` instead?

Reply:
Tried it, I'll find:
```go
        if strings.ContainsFunc(s, func(r rune) bool { return r > unicode.MaxASCII }) {
            return LocalizableError("has unescaped non-ASCII characters")
        }
```
slightly less readable (but this may be a matter of not being accustomed to it) than your original code from https://github.com/gocsaf/csaf/pull/517 . Is it more efficient?